modules/persistent-storage-csi-snapshots-delete.adoc

```diff
@@ -26,4 +25,46 @@ deletionPolicy: Delete <1>
 ----
 <1> If the `Delete` value is set, the underlying snapshot will be deleted, along with the `VolumeSnapshotContent` object. If the `Retain` value is set, both the underlying snapshot and `VolumeSnapshotContent` object remain.
   +
-If the `Retain` value is set, and the `VolumeSnapshot` object is deleted without deleting the corresponding `VolumeSnapshotContent` object, then the content will remain. The snapshot itself is also retained in the storage back end.
+If the `Retain` value is set and the `VolumeSnapshot` object is deleted without deleting the corresponding `VolumeSnapshotContent` object, the content will remain. The snapshot itself is also retained in the storage back end.
+
+. Optional: If the `VolumeSnapshot` object is not successfully deleted, enter the following commands to remove the existing finalizers from the `VolumeSnapshot` object so that the delete operation can continue:
++
+[IMPORTANT]
+====
+Only remove the finalizers if you are confident that there are no existing references from either persistent volume claims or volume snapshot contents to the `VolumeSnapshot` object.
+
+Even with the `--force` option, the delete operation does not delete snapshot objects until all finalizers are removed.
+====
+
+.. Remove the finalizer for the leftover resources by patching the finalizer with null:
++
+[source,terminal]
+----
+$ oc patch -n $PROJECT volumesnapshot/$NAME --type=merge -p '{"metadata": {"finalizers":null}}'
+----
+
+.. Delete the volume snapshot class:
```

Comment:
@bobfuru @huffmanca I'm not sure if volumesnapshotclass deletion here is expected? Or do you want to VolumeSnapshotContent instead?

Reply:
This step shouldn't be needed at all. It can be removed.